Source/monster.cpp

```diff
@@ -55,9 +55,8 @@ size_t LevelMonsterTypeCount;
 Monster Monsters[MaxMonsters];
 int ActiveMonsters[MaxMonsters];
 size_t ActiveMonsterCount;
-// BUGFIX: replace MonsterKillCounts[MaxMonsters] with MonsterKillCounts[NUM_MTYPES].
 /** Tracks the total number of monsters killed per monster_id. */
-int MonsterKillCounts[MaxMonsters];
+int MonsterKillCounts[NUM_MTYPES];

```

Comment:
You need to make sure the save file skips/pads the difference.
https://github.com/diasurgical/devilutionX/blob/c534e0e93c07d6d6d85151eeccac1d5699b988da/Source/loadsave.cpp#L2096-L2097
https://github.com/diasurgical/devilutionX/blob/c534e0e93c07d6d6d85151eeccac1d5699b988da/Source/loadsave.cpp#L2350-L2351